File: backend/model/application.go
/**
 * @Author: Nan
 * @Date: 2024/10/23 21:47
 */

package model

type Application struct {
	Basic
	Name       string `json:"name" gorm:"type:varchar(255);not null;comment:应用名称"`
	AppKey     string `json:"app_key" gorm:"type:varchar(255);not null;comment:应用key"`
	AppSecret  string `json:"app_secret" gorm:"type:varchar(255);not null;comment:应用secret"`
	Expiration int64  `json:"expiration" gorm:"type:int;not null;comment:过期时间"`
	DingKey    string `json:"ding_key" gorm:"type:varchar(255);comment:钉钉key"`
	DingSecret string `json:"ding_secret" gorm:"type:varchar(255);comment:钉钉secret"`
	DingAppID  string `json:"ding_app_id" gorm:"column:ding_app_id;type:varchar(255);comment:钉钉appID"`
	Remark     string `json:"remark" gorm:"type:varchar(255);comment:备注"`
}

File: frontend/src/api/services/application.ts
import type { Basic, Query, ResponseData } from 'src/types/global'
import { instance } from 'boot/axios'

export interface Application extends Basic {
  name: string
  app_key: string
  expiration: number
  ding_key: string
  ding_app_id: string
  remark: string
}

export interface ApplicationCreateReq {
  name: string
  expiration: number
  ding_key: string
  ding_secret: string
  ding_app_id: string
  remark: string
}

export interface ApplicationUpdateReq {
  name: string
  expiration: number
  ding_key: string
  ding_secret: string
  ding_app_id: string
  remark: string
}

export interface ApplicationSecretRes {
  id: number
  name: string
  app_key: string
  app_secret: string
  expiration: number
}

export const useApplicationApi = () => {
  const queryApplication = async (params: Query) => {
    return instance
      .post<ResponseData<Array<Application>>>('/admin/application/query', params)
      .then((res) => {
        return res.data
      })
  }

  const queryApplicationById = (id: number) => {
    return instance.get<ResponseData<Application>>(`/admin/application/${id}`).then((res) => {
      return res.data
    })
  }

  const createApplication = async (req: ApplicationCreateReq) => {
    return instance.post<ResponseData<ApplicationSecretRes>>('/admin/application', req).then((res) => {
      return res.data
    })
  }

  const updateApplication = async (id: number, req: ApplicationUpdateReq) => {
    return instance.put<ResponseData<BigInteger>>(`/admin/application/${id}`, req).then((res) => {
      return res.data
    })
  }

  const deleteApplication = async (id: number) => {
    return instance.delete<ResponseData<BigInteger>>(`/admin/application/${id}`).then((res) => {
      return res.data
    })
  }

  const rotateApplicationSecret = async (id: number) => {
    return instance
      .post<ResponseData<ApplicationSecretRes>>(`/admin/application/${id}/rotate-secret`)
      .then((res) => {
        return res.data
      })
  }

  return {
    queryApplication,
    queryApplicationById,
    createApplication,
    updateApplication,
    deleteApplication,
    rotateApplicationSecret,
  }
}
